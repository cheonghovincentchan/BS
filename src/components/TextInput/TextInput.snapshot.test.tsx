/**
 * snapshots unit testing docs https://jestjs.io/docs/tutorial-react#snapshot-testing
 */
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

jest.mock('@ant-design/icons/ExclamationCircleFilled', () => () => {
  return <span data-testid="ExclamationCircleFilled" />;
});
jest.mock('@ant-design/icons/CheckCircleFilled', () => () => {
  return <span data-testid="CheckCircleFilled" />;
});

describe('TextInput snapshot unit test', () => {
  it('bacic test', () => {
    const component = renderer.create(<TextInput />);
    const tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
    renderer.act(() => {
      tree.props.status = 'success';
    });
  });
  it('status test', () => {
    const component = renderer.create(<TextInput status="success" helpText="helptext" />);
    const tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

    const component2 = renderer.create(<TextInput status="error" helpText="helptext" />);
    const tree2 = component2.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree2).toMatchSnapshot();

    const component3 = renderer.create(<TextInput status="warning" helpText="helptext" />);
    const tree3 = component3.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree3).toMatchSnapshot();
  });
});
