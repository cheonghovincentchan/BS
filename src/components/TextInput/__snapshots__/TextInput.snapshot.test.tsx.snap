// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot unit test bacic test 1`] = `
<section>
  <input
    className="ant-input css-dev-only-do-not-override-ixblex h-[1.15rem] text-[0.57rem]"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#D5D4DC",
        "fontSize": "0.386rem",
        "width": "7.9rem",
      }
    }
    type="text"
    value=""
  />
  <div
    className="flex items-center"
  >
    <span
      className="text-[0.32rem] ml-[0.1rem]"
      style={
        {
          "color": "#D5D4DC",
        }
      }
    />
  </div>
</section>
`;

exports[`TextInput snapshot unit test status test 1`] = `
<section>
  <input
    className="ant-input css-dev-only-do-not-override-ixblex h-[1.15rem] text-[0.57rem]"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      {
        "backgroundColor": "#F1F8F2",
        "borderColor": "#2BAC47",
        "fontSize": "0.386rem",
        "width": "7.9rem",
      }
    }
    type="text"
    value=""
  />
  <div
    className="flex items-center"
  >
    <span
      data-testid="CheckCircleFilled"
    />
    <span
      className="text-[0.32rem] ml-[0.1rem]"
      style={
        {
          "color": "#2BAC47",
        }
      }
    >
      helptext
    </span>
  </div>
</section>
`;

exports[`TextInput snapshot unit test status test 2`] = `
<section>
  <input
    className="ant-input css-dev-only-do-not-override-ixblex h-[1.15rem] text-[0.57rem]"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      {
        "backgroundColor": "#FBEFEF",
        "borderColor": "#C83532",
        "fontSize": "0.386rem",
        "width": "7.9rem",
      }
    }
    type="text"
    value=""
  />
  <div
    className="flex items-center"
  >
    <span
      data-testid="ExclamationCircleFilled"
    />
    <span
      className="text-[0.32rem] ml-[0.1rem]"
      style={
        {
          "color": "#C83532",
        }
      }
    >
      helptext
    </span>
  </div>
</section>
`;

exports[`TextInput snapshot unit test status test 3`] = `
<section>
  <input
    className="ant-input css-dev-only-do-not-override-ixblex h-[1.15rem] text-[0.57rem]"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      {
        "backgroundColor": "#FDF3EC",
        "borderColor": "#EF8943",
        "fontSize": "0.386rem",
        "width": "7.9rem",
      }
    }
    type="text"
    value=""
  />
  <div
    className="flex items-center"
  >
    <span
      data-testid="ExclamationCircleFilled"
    />
    <span
      className="text-[0.32rem] ml-[0.1rem]"
      style={
        {
          "color": "#EF8943",
        }
      }
    >
      helptext
    </span>
  </div>
</section>
`;
